crates/rover-client/src/operations/cloud/config/fetch.rs

```diff
@@ -0,0 +1,96 @@
+use super::types::{CloudConfigFetchInput, CloudConfigFetchResponse};
+
+use graphql_client::*;
+
+use crate::blocking::StudioClient;
+use crate::shared::GraphRef;
+use crate::RoverClientError;
+
+#[derive(GraphQLQuery, Debug)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/cloud/config/fetch_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "Eq, PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+pub struct CloudConfigFetchQuery;
+
+pub fn run(
+    input: CloudConfigFetchInput,
+    client: &StudioClient,
+) -> Result<CloudConfigFetchResponse, RoverClientError> {
+    let graph_ref = input.graph_ref.clone();
+    let data = client.post::<CloudConfigFetchQuery>(input.into())?;
+    build_response(graph_ref, data)
+}
+
+fn build_response(
+    graph_ref: GraphRef,
+    data: cloud_config_fetch_query::ResponseData,
+) -> Result<CloudConfigFetchResponse, RoverClientError> {
+    let variant = data
+        .graph
+        .ok_or(RoverClientError::GraphNotFound {
+            graph_ref: graph_ref.clone(),
+        })?
+        .variant
+        .ok_or(RoverClientError::GraphNotFound {
+            graph_ref: graph_ref.clone(),
+        })?;
+
+    let config = variant
+        .router_config
+        .ok_or(RoverClientError::MalformedResponse {
+            null_field: "router_config".to_string(),
+        })?;
+
+    Ok(CloudConfigFetchResponse { graph_ref, config })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::shared::GraphRef;
+    use serde_json::json;
+
+    fn mock_graph_ref() -> GraphRef {
+        GraphRef {
+            name: "mygraph".to_string(),
+            variant: "current".to_string(),
+        }
+    }
```

Comment:
Can we use `rstest` here instead and turn this into a fixture?

Reply:
I tried rtest and it made it more fiddly. I've added assertions for the error types 👍